components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/AuthenticationValve.java

```diff
@@ -178,15 +167,31 @@ public void invoke(Request request, Response response) throws IOException, Servl
             unsetThreadLocalAuthUserTenantDomain();
             // Clear thread local provisioning service provider.
             IdentityApplicationManagementUtil.resetThreadLocalProvisioningServiceProvider();
-            // Clear clientComponent in MDC.
-            unsetClientComponent();
+            // Clear Thread Locals from MDC.
+            unsetMDCThreadLocals();
             // Clear thread local authenticated with basic auth flag.
             unsetAuthenticatedWithBasicAuth();
         }
 
 
     }
 
+    private void setRemoteAddressAndUserAgentToMDC(Request request){
+
+        String userAgent = request.getHeader(USER_AGENT);
+        String forwardedUserAgent = request.getHeader(X_FORWARDED_USER_AGENT);
+        if (StringUtils.isNotEmpty(forwardedUserAgent)) {
+            userAgent = forwardedUserAgent;
+        }
+        String remoteAddr = request.getRemoteAddr();
+        if (StringUtils.isNotEmpty(userAgent) && isLoggableParam(CONFIG_LOG_PARAM_USER_AGENT)) {
+            MDC.put(USER_AGENT, request.getHeader(USER_AGENT));
```

Comment:
should we use the `userAgent` here?

Reply:
@Shaaali @senthalan  If the `X_FORWARDED_USER_AGENT` is available, shouldn't it be added to the MDC? I mean can't we use the `userAgent` variable here.